Source/URLProtocol.swift

```diff
@@ -195,29 +199,41 @@ class URLProtocol: NSURLProtocol, NSURLConnectionDataDelegate {
             client?.URLProtocol(self, cachedResponseIsValid: cachedResponse)
             return
         }
-        self.connection = NSURLConnection(request: mutableRequest, delegate: self)
+        self.dataTask = self.session.dataTaskWithRequest(request)
+        self.dataTask?.resume()
     }
 
     override func stopLoading() {
-        connection?.cancel()
-        connection = nil
+        self.dataTask?.cancel()
+        self.dataTask = nil
     }
 
-    // Mark: - NSURLConnectionDataDelegate Methods
+    // Mark: - NSURLSessionDataDelegate Methods
 
-    func connection(connection: NSURLConnection, didReceiveResponse response: NSURLResponse) {
+    func URLSession(session: NSURLSession,
+        dataTask: NSURLSessionDataTask,
+        didReceiveResponse response: NSURLResponse,
+        completionHandler: (NSURLSessionResponseDisposition) -> Void)
+    {
+        completionHandler(.Allow)
         self.client?.URLProtocol(self, didReceiveResponse: response, cacheStoragePolicy: .Allowed)
     }
 
-    func connection(connection: NSURLConnection, didReceiveData data: NSData) {
+    func URLSession(session: NSURLSession,
+        dataTask: NSURLSessionDataTask,
+        didReceiveData data: NSData)
+    {
         self.client?.URLProtocol(self, didLoadData: data)
     }
 
-    func connectionDidFinishLoading(connection: NSURLConnection) {
-        self.client?.URLProtocolDidFinishLoading(self)
-    }
-
-    func connection(connection: NSURLConnection, didFailWithError error: NSError) {
-        self.client?.URLProtocol(self, didFailWithError: error)
+    func URLSession(session: NSURLSession,
+        task: NSURLSessionTask,
+        didCompleteWithError error: NSError?)
+    {
+        if let error = error {
+            self.client?.URLProtocol(self, didFailWithError: error)
+        } else {
+            self.client?.URLProtocolDidFinishLoading(self)
```

Comment:
Nah, that means it completed successfully if there's no error. There's just a single delegate method for both cases instead of the two separate methods.